### PR TITLE
freebsd: Don't assume that root user is in root group

### PR DESCRIPTION
**Pull request by holmanb**
```
Don't assume root user is in root group

BSD's root user has group name "wheel" for GID 0

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265177
```

Rather than assuming that "root:root" is default user:group everywhere, move this value into the distro definitions as the default, and override this to "root:wheel" for ~~freebsd~~ BSDs.

I also added some type annotations along the way, which help us catch/prevent bugs thanks to mypy and friends.

### CODE REVIEW

**Comment by holmanb** [comment]
Ah, it looks like I have freebsd source local from the last time I looked at a freebsd bug. I see:
```
arc~/workspace/freebsd-src(main|✔) grep root etc/group
wheel:*:0:root
operator:*:5:root
```

**Comment by holmanb** [comment]
@igalic do you have time for a quick review? I'd like a review from a freebsd user or developer.

File: cloudinit/distros/freebsd.py
Comment (igalic) [inline cloudinit/distros/freebsd.py:29]: this is the case for all BSDs, so this can go into bsd.py

**Review by igalic — CHANGES_REQUESTED** [review]
minor change for a big impact ;)

**Comment by holmanb** [comment]
@igalic Thanks for the review! It should be good now I think.

**Review by TheRealFalcon — APPROVED** [review]
LGTM, though I won't merge right now to see if we have anymore comments.

**Comment by holmanb** [comment]
Thanks for the reviews @igalic and @TheRealFalcon! Merging now.

**Pull request merged**
